src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/validator/UpdatePlanExemptionReasonConstraintValidator.kt

```diff
@@ -24,11 +24,7 @@ class UpdatePlanExemptionReasonConstraintValidator : ConstraintValidator<ReasonS
     val request = parameters.find { it is UpdatePlanCreationStatusRequest } as? UpdatePlanCreationStatusRequest
       ?: return false // parameter wasn't found, this will be already have been trapped by the controller.
 
-    return if (request.status in exemptStatuses()) {
```

Comment:
Looking at this validator, how it was called, and what it does, I think actually the notes I put in the ticket might not be right / the best way to do this ....

I wonder if we need to revert the changes to the swagger spec, to make exemptionReason and exemptionDetail optional again, and do the validation here, like this:

```
    return if (request.status in exemptStatuses()) {
      request.exemptionReason != null && request.exemptionDetail != null
    } else {
      request.exemptionReason == null && request.exemptionDetail == null
    }
```

This means that we add additional statuses to the `UpdatePlanCreationStatusRequest.status` object in the future, that may or may not require these 2 exemption fields

Does this make sense? What do you think? 🤔